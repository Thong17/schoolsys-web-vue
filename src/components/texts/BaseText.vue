<template>
  <p class="text-gray-700G" :class="`text-${size}`" :style="{ color }">
    <slot />
  </p>
</template>

<script lang="ts">
  export default {
    name: 'BaseText',
    props: {
      size: {
        type: String,
        validator: (value: 'md' | 'sm' | 'lg') =>
          ['md', 'sm', 'lg'].includes(value),
        default: 'md',
      },
      color: {
        type: String,
        default: 'gray',
      },
    },
  }
</script>

<style></style>
