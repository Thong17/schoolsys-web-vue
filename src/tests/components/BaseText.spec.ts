import { expect, it } from 'vitest'
import BaseText from '@/components/texts/BaseText.vue'
import { mount } from '@vue/test-utils'

it.each(['sm', 'md', 'lg'])(
  'Base Text should render text Hello World',
  async (size) => {
    const wrapper = mount(BaseText, {
      slots: { default: 'Hello World' },
      props: { size },
    })
    expect(wrapper.text()).toBe('Hello World')
    expect(wrapper.html()).toMatchSnapshot()
  }
)
