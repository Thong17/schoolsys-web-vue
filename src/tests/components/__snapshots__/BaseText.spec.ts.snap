// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base Text should render text Hello World 1`] = `"<p class="text-gray-700G text-sm" style="color: gray;">Hello World</p>"`;

exports[`Base Text should render text Hello World 2`] = `"<p class="text-gray-700G text-md" style="color: gray;">Hello World</p>"`;

exports[`Base Text should render text Hello World 3`] = `"<p class="text-gray-700G text-lg" style="color: gray;">Hello World</p>"`;
